drivers/flash/flash_stm32_ospi.c

```diff
@@ -2002,8 +2046,29 @@ static int spi_nor_process_bfp(const struct device *dev,
 		++etp;
 	}
 
-	spi_nor_process_bfp_addrbytes(dev, jesd216_bfp_addrbytes(bfp));
+	addr_mode = jesd216_bfp_addrbytes(bfp);
+	spi_nor_process_bfp_addrbytes(dev, addr_mode);
 	LOG_DBG("Address width: %u Bytes", data->address_width);
+	/* 4 Byte Address Mode has to be explicitly enabled for Winbond Flash */
+	if (addr_mode == JESD216_SFDP_BFP_DW1_ADDRBYTES_VAL_3B4B) {
+		struct jesd216_bfp_dw16 dw16;
+
+		if (jesd216_bfp_decode_dw16(php, bfp, &dw16) == 0) {
+			/*
+			 * According to JESD216, the bit0 of dw16.enter_4ba
+			 * portion of flash description register 16 indicates
+			 * if it is enough to use 0xB7 instruction without
+			 * write enable to switch to 4 bytes addressing mode.
+			 * If bit 1 is set, a write enable is needed.
+			 */
+			if (dw16.enter_4ba & 0x3) {
```

Comment:
Shouldn't here be '& 0x01'? Currently this will also enter on 'enter_4ba == 0x02', which would not confirm that command is supported.

Reply:
Current logic is fine.

For `dw16.enter_4ba`:

* If `1` - issue instruction `0xB7` (_write enable `0x06` is not required_)
* If `2` - issues write enable `0x06`, then `0xB7`

Reply:
keeping "dw16.enter_4ba & 0x3"